tests/goto_wcc_invalid_test.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+set -o pipefail
+
+source ./test_sub.sh
+
+function test_backward_goto() {
+  begin_test_suite "Backward Goto"
+  
+  compile_error 'simple backward goto' '
+int main() {
+start_label:
+    int x = 1;
+    if (x) {
+        goto start_label;  // BACKWARD GOTO - should fail
+    }
+    return 0;
+}'
+
+  compile_error 'backward goto in nested loops' '
+int main() {
```

Comment:
You may remove this test as I feel it is a duplication of what you are checking.